simulator_api/rest_server/exposed_methods.py

```diff
@@ -0,0 +1,75 @@
+
+import json
+
+from flask import Blueprint, request
+from WebServerCore.command_factory import CommandFactorySingleton
+from WebServerCore.handler import handler_get, handler_post, handler_put
+
+# The handler functions below expose the endpoints.
+# They are necessary for the application to work, in this case, with the Flask framework.
+# The logic for each function is implemented by its namesake on the factory side.
+
+commands = Blueprint("commands", __name__)
+
+
+@commands.before_app_first_request
+def discovery():
+    """Lazy instantiation of the commands discovery. Commands will be discovered on first call."""
+    CommandFactorySingleton.discover_commands(__file__)
+
+
+@commands.route("/api/<version>/<get_method>", methods=["GET"])
+def get_call(version, get_method):
+    """Forward the http get calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_get(get_method, request, version)
+
+@commands.route("/api/<get_method>", methods=["GET"])
+def get_call_no_version(get_method):
+    """Forward the http get calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_get(get_method, request)
+
+@commands.route("/api/<version>/<get_method>/<task_id>", methods=["GET"])
+def get_status_call(version, get_method, task_id):
+    """Forward the http get calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_get(get_method, request, version, task_id)
+
+@commands.route("/api/<get_method>/<task_id>", methods=["GET"])
+def get_status_call_no_version(get_method, task_id):
+    """Forward the http get calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_get(get_method, request, task_id=task_id)
+
+
+@commands.route("/api/<version>/<post_method>", methods=["POST"])
+def post_call(version, post_method):
+    """Forward the http post calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_post(post_method, request, version)
+
+@commands.route("/api/<post_method>", methods=["POST"])
+def post_call_no_version(post_method):
+    """Forward the http post calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_post(post_method, request)
+
+@commands.route("/api/<version>/<put_method>", methods=["PUT"])
+def put_call(version, put_method):
+    """Forward the http put calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_put(put_method, request, version)
+
+@commands.route("/api/<put_method>", methods=["PUT"])
+def put_call_no_version(put_method):
+    """Forward the http put calls to the WebServer core handler to take advantage of the framework functionalities"""
+    return handler_put(put_method, request)
+
+
+# Test function.
+@commands.route("/")
```

Comment:
maybe we can remove test function

Reply:
the test function is used to inform the user of function get-capabilities